packages/ldap-auth-backend/README.md

```diff
@@ -1 +1,187 @@
+import { ScaffolderEntitiesProcessor } from '@backstage/plugin-scaffolder-backend';
+import {
+  LdapOrgEntityProvider,
+} from '@backstage/plugin-catalog-backend-module-ldap';
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  const builder = await CatalogBuilder.create(env);
+
+  builder.addEntityProvider(
+    LdapOrgEntityProvider.fromConfig(env.config, {
+      id: '<YOUR-ID>',
+      target: 'ldaps://<YOUR-ADDRESS>',
+      logger: env.logger,
+      schedule: env.scheduler.createScheduledTaskRunner({
+        frequency: // whatever
+        timeout: // whatever
+      }),
+    }),
+  );
+
+  builder.addProcessor(new ScaffolderEntitiesProcessor());
+  const { processingEngine, router } = await builder.build();
+  await processingEngine.start();
+  return router;
+}
+```
+
+### Add authentication backend
+
+This assumes a basic usage: single process without custom auth function or user object customization and in-memory token storage
+
+For more uses cases you can see the [example folders](../../examples/)
+
+> `packages/backend/src/plugins/auth.ts`
+
+```ts
+import { createRouter } from '@backstage/plugin-auth-backend';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
+import { ldap } from '@immobiliarelabs/backstage-plugin-ldap-auth-backend';
+
+export default async function createPlugin(
+    env: PluginEnvironment
+): Promise<Router> {
+    return await createRouter({
+        logger: env.logger,
+        config: env.config,
+        database: env.database,
+        discovery: env.discovery,
+        tokenManager: env.tokenManager,
+        providerFactories: {
+            ldap: ldap.create({}),
+        },
+    });
+}
+```
+
+### Add the login form
+
+> More on this in the frontend plugin documentation [here](../ldap-auth/README.md)
+
+We need to replace the existing Backstage demo authentication page with our custom one!
+
+In the `App.tsx` file change the `createApp` function to provide use our custom `SignInPage` in the `components` key.
+
+**Note:** This components isn't only UI, it also brings all the token state management and HTTP API calls to the backstage auth routes we already configured in the backend part.
+
+> `packages/app/src/App.tsx`
+
+```tsx
+import { LdapAuthFrontendPage } from '@immobiliarelabs/backstage-plugin-ldap-auth';
+
+const app = createApp({
+    // ...
+    components: {
+        SignInPage: (props) => (
+            <LdapAuthFrontendPage {...props} provider="ldap" />
+        ),
+    },
+    // ...
+});
+```
+
+And you're ready to go! If you need more use cases, like having multiple processes and need a shared token store instead of in-memory look at the [examples folder](https://github.com/immobiliare/backstage-plugin-ldap-auth/examples)
+
+## Powered Apps
+
+dats was created by the amazing Node.js team at ImmobiliareLabs, the Tech dept of [Immobiliare.it](https://www.immobiliare.it), the #1 real estate company in Italy.
```

Comment:
`[ImmobiliareLabs](https://labs.immobiliare.it)`

Reply:
Also here there is `dats` word!